specification/security/resource-manager/Microsoft.Security/preview/2023-05-01-preview/customRecommedations.json

```diff
@@ -0,0 +1,384 @@
+          "$ref": "#/definitions/CustomRecommendationProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "CustomRecommendationProperties": {
+      "type": "object",
+      "description": "describes the Custom Recommendation properties",
+      "properties": {
+        "query": {
+          "description": "KQL query representing the Recommendation results required.",
+          "type": "string"
+        },
+        "cloudProviders": {
+          "description": "List of all standard supported clouds.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RecommendationSupportedClouds"
+          }
+        },
+        "severity": {
+          "description": "The severity to relate to the assessments generated by this Recommendation.",
+          "type": "string",
+          "enum": [
+            "High",
+            "Medium",
+            "Low"
+          ],
+          "x-ms-enum": {
+            "name": "severityEnum",
+            "modelAsString": true
+          },
+          "x-ms-client-default": "Low"
+        },
+        "displayName": {
+          "description": "The display name of the assessments generated by this Recommendation.",
+          "type": "string"
+        },
+        "description": {
+          "description": "The description to relate to the assessments generated by this Recommendation.",
+          "type": "string"
+        },
+        "remediationDescription": {
+          "description": "The remediation description to relate to the assessments generated by this Recommendation.",
+          "type": "string"
+        },
+        "assessmentKey": {
+          "description": "The assessment metadata key used when an assessment is generated for this Recommendation.",
+          "type": "string",
+          "readOnly": true
+        },
+        "metadata": {
+          "$ref": "#/definitions/CustomRecommendationMetaData"
+        }
+      }
+    },
+    "CustomRecommendationMetaData": {
+      "type": "object",
+      "description": "The recommendation metadata",
+      "properties": {
+        "createdBy": {
+          "description": "recommendation Created by object id (GUID)",
+          "type": "string",
+          "readOnly": true
+        },
+        "createdOn": {
+          "description": "recommendation creation date",
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true
+        },
+        "updatedBy": {
+          "description": "recommendation last updated by object id (GUID)",
+          "type": "string",
+          "readOnly": true
+        },
+        "updatedOn": {
+          "description": "recommendation last update date",
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true
+        }
+      }
+    },
+    "RecommendationSupportedClouds": {
```

Comment:
>RecommendationSupportedClouds [](http://example.com/codeflow?start=5&length=29)

This should be singular, as it can be just one of the cloud providers.